### PR TITLE
E0d/diverse virtualization targets

### PR DESCRIPTION
**Pull request by e0d**
This PR puts the structure in place for refactoring virtualization target specific code and moves an AWS specific task.

### CODE REVIEW

File: playbooks/roles/vhost/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#
+# 
+# Example play:
+#
+#
+
+- name: add ssh-warning banner motd
+  template: >
+    dest=/etc/motd.tail
+    src={{ COMMON_MOTD_TEMPLATE }} mode=0755 owner=root group=root
```
Comment (feanil) [inline playbooks/roles/vhost/tasks/main.yml:27]: Shouldn't this template get pulled out as well?

File: playbooks/roles/vhost/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+- name: update ssh config
+  template: >
+    dest=/etc/ssh/sshd_config
+    src=sshd_config.j2 mode=0644 owner=root group=root
```
Comment (feanil) [inline playbooks/roles/vhost/tasks/main.yml:32]: This one will need to be moved into the vhost role.

**Comment by feanil** [comment]
I didn't see anything calling the vhost role, do there need to be updates to our playbooks?

**Comment by e0d** [comment]
@feanil yes, if we are happy with the direction, the playbooks will need to be updated.

**Comment by e0d** [comment]
@feanil @maxrothman @fredsmith I think it may make sense to merge this into the branch and fix forward, objections?

**Comment by maxrothman** [comment]
I'd be comfortable merging the new files, but I'm uncomfortable with merging the modifications to existing files without actual testing.

EDIT: @e0d wait, did you mean a branch other than master?

**Pull request closed without merging**